docs/openfaas-pro/iam/function-authentication.md

```diff
@@ -0,0 +1,275 @@
+# Function Authentication
+
+With OpenFaaS Identity and Access Management (IAM) you can secure function endpoints without having to write any additional code. Access permissions for functions can be configured using IAM Roles and Policies.
+
+!!! note Technical preview
+    Function authentication is currently under technical preview
+
+## Enable function authentication
+
+OpenFaaS Identity and Access Management needs to be enabled and configured in order to use function authentication, see: [Identity and Access Management (IAM)](/openfaas-pro/iam/overview/#installation)
+
+To enable function authentication your functions need to be using a compatible version of the [OpenFaaS watchdog](/architecture/watchdog/). For the classic-watchdog this is any version `> 0.2.3`. The of-watchdog support function authentication for releases `> 0.9.15`.
+
+Authentication can be enabled on a per function basis by setting the `jwt_auth` environment variable to true in the function configuration, e.g:
+
+```yaml
+functions:
+    figlet:
+        skip_build: true
+        image: ghcr.io/openfaas/figlet:latest
+        environment:
+            jwt_auth: true
+```
+
```

Comment:
Some note saying that existing functions are not affected, and functions without the jwt_auth environment variable will not be affected.

Reply:
A note has been added for this.